src/formatDate.js

```diff
@@ -45,12 +45,67 @@
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
- *
+ *formatDate.js
  * @returns {string}
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newFormatDate = '';
+  let newDay = '';
+  let newMonth = '';
+  let newYear = '';
+  const newSeparator = toFormat[3];
+
+  let oldDateSplit = [];
+  let oldYear = '';
+
+  oldDateSplit = date.split(fromFormat[3]);
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'DD') {
+      newDay = oldDateSplit[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      newMonth = oldDateSplit[i];
+    }
+
+    if (fromFormat[i].slice(0, 2) === 'YY') {
+      oldYear = oldDateSplit[i];
+    }
+  }
+
+  for (const elem of toFormat) {
+    if (elem === 'DD') {
+      newFormatDate += (newDay + newSeparator);
+    }
+
+    if (elem === 'MM') {
+      newFormatDate += (newMonth + newSeparator);
+    }
+
+    if (elem.slice(0, 2) === 'YY') {
+      if (elem.length === oldYear.length) {
+        newYear = oldYear;
+      }
+
+      if (elem.length < oldYear.length) {
+        newYear = oldYear.slice(2, 4);
+      }
+
+      if (elem.length > oldYear.length) {
+        if (oldYear < '30') {
+          newYear = '20' + oldYear;
+        } else {
+          newYear = '19' + oldYear;
+        }
+      }
+
+      newFormatDate += (newYear + newSeparator);
+    }
+  }
+
+  return newFormatDate.slice(0, (newFormatDate.length - 1));
```

Comment:
The 'slice' method is used to remove the last separator. Consider using the 'join' method to concatenate date parts, it will automatically add separators between elements and avoid an extra removal operation.